### PR TITLE
[#506] Removed reflection call.

### PR DESCRIPTION
**Pull request by ArneDeutsch**
Do not inherit and modify (by reflection) the original
java.util.jar.Manifest but instead C&P and modify the code. Adapt
ManifestMergerTest by importing the new classes.

Signed-off-by: Arne Deutsch <Arne.Deutsch@itemis.de>

### CODE REVIEW

**Comment by ArneDeutsch** [comment]
@szarnekow Sorry for the mess. Please review again. This time the PR should be ok.

File: org.eclipse.xtext.util/src/org/eclipse/xtext/util/MergeableManifest.java [hunk 1]
```diff
@@ -378,17 +382,21 @@ public void addImportedPackages(Set<String> packages) {
 	}
 
 	/**
-	 * @deprecated Use {@link #mergeIntoCommaSeparatedList(String, Set, Wrapper, String)} instead.
+	 * @deprecated Use
```
Comment (szarnekow) [inline org.eclipse.xtext.util/src/org/eclipse/xtext/util/MergeableManifest.java]: You evil code formatter.

File: org.eclipse.xtext.util/src/org/eclipse/xtext/util/MergeableManifest.java [hunk 2]
```diff
@@ -418,30 +426,952 @@ public static String mergeIntoCommaSeparatedList(String currentString, Set<Strin
...
+		return attr.hashCode() + entries.hashCode();
+	}
+
+	/**
+	 * Returns a shallow copy of this Manifest. The shallow copy is implemented
+	 * as follows:
+	 * 
+	 * <pre>
+	 * public Object clone() {
+	 * 	return new Manifest(this);
```
Comment (szarnekow) [inline org.eclipse.xtext.util/src/org/eclipse/xtext/util/MergeableManifest.java]: Outdated copied comment

File: org.eclipse.xtext.util/src/org/eclipse/xtext/util/MergeableManifest.java [hunk 3]
```diff
@@ -117,8 +131,8 @@ public Object put(Object name, Object value) {
 		}
 
 		/*
-		 * Copied from base class, but replaced call to make72Safe(buffer) with make512Safe(buffer)
-         * and does not write empty values
+		 * Copied from base class, but replaced call to make72Safe(buffer) with
```
Comment (szarnekow) [inline org.eclipse.xtext.util/src/org/eclipse/xtext/util/MergeableManifest.java]: Which base class?

File: org.eclipse.xtext.util/src/org/eclipse/xtext/util/MergeableManifest.java [hunk 4]
```diff
@@ -11,30 +11,44 @@
...
-import java.util.jar.Attributes;
 import java.util.jar.Manifest;
 
+import org.apache.log4j.Logger;
+
 import com.google.common.base.Objects;
 import com.google.common.collect.Lists;
 
 /**
+ * Code is largely C&P from {@link Manifest}. Needed to adapt some details (line
```
Comment (kthoms) [inline org.eclipse.xtext.util/src/org/eclipse/xtext/util/MergeableManifest.java]: The & sign is here in Javadoc and must be escaped, or just use "copied"

File: org.eclipse.xtext.util/src/org/eclipse/xtext/util/MergeableManifest.java [hunk 5]
```diff
@@ -11,30 +11,44 @@
...
+ * length, detection or changes when only ordering is changed). Original code
+ * has extended {@link Manifest} and used reflection to modify behavior. This
+ * gives warnings with Java 9/10 and will probably fail with future versions.
+ * Hence it is better to have some C&P code instead of the reflection warning.
```
Comment (kthoms) [inline org.eclipse.xtext.util/src/org/eclipse/xtext/util/MergeableManifest.java]: same here with &

**Comment by kthoms** [comment]
Sorry for nitpicking, please update with my new comments.

**Comment by kthoms** [comment]
Thanks, @ArneDeutsch

**Pull request merged**